auto-value-gson/src/test/java/com/ryanharter/auto/value/gson/AutoValueGsonExtensionTest.java

```diff
@@ -143,62 +142,33 @@ public void simple() {
-        + "Number> f,\n"
-        + "      Set<String> g, Map<String, Set<String>> h, String i, @Nullable List<String> j, String k, "
-        + "List<String> l,\n"
-        + "      String m, List<String> n,\n"
+        + "  AutoValue_Test(String a, int[] b, int c, String d, @Nullable String e,\n"
+        + "      ImmutableMap<String, Number> f, Set<String> g, Map<String, Set<String>> h, String i,\n"
+        + "      @Nullable List<String> j, String k, List<String> l, String m, List<String> n,\n"
         + "      Map<String, Map<String, Map<String, Map<String, Map<String, String>>>>> o) {\n"
         + "    super(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o);\n"
         + "  }\n"
         + "\n"
         + "  public static final class GsonTypeAdapter extends TypeAdapter<Test> {\n"
-        + "    private final TypeAdapter<String> aAdapter;\n"
-        + "    private final TypeAdapter<int[]> bAdapter;\n"
-        + "    private final TypeAdapter<Integer> cAdapter;\n"
-        + "    private final TypeAdapter<String> dAdapter;\n"
-        + "    private final TypeAdapter<String> eAdapter;\n"
-        + "    private final TypeAdapter<ImmutableMap<String, Number>> fAdapter;\n"
-        + "    private final TypeAdapter<Set<String>> gAdapter;\n"
-        + "    private final TypeAdapter<Map<String, Set<String>>> hAdapter;\n"
-        + "    private final TypeAdapter<String> iAdapter;\n"
-        + "    private final TypeAdapter<List<String>> jAdapter;\n"
-        + "    private final TypeAdapter<String> kAdapter;\n"
-        + "    private final TypeAdapter<List<String>> lAdapter;\n"
-        + "    private final TypeAdapter<String> mAdapter;\n"
-        + "    private final TypeAdapter<List<String>> nAdapter;\n"
-        + "    private final TypeAdapter<Map<String, Map<String, Map<String, Map<String, "
-        + "Map<String, String>>>>>> oAdapter;\n"
+        + "    private final TypeAdapter<String> java_lang_StringAdapter;\n"
+        + "    private final TypeAdapter<int[]> int__Adapter;\n"
+        + "    private final TypeAdapter<Integer> int_Adapter;\n"
+        + "    private final TypeAdapter<ImmutableMap<String, Number>> com_google_common_collect_ImmutableMap_java_lang_String__java_lang_Number_Adapter;\n"
+        + "    private final TypeAdapter<Set<String>> java_util_Set_java_lang_String_Adapter;\n"
+        + "    private final TypeAdapter<Map<String, Set<String>>> java_util_Map_java_lang_String__java_util_Set_java_lang_String__Adapter;\n"
+        + "    private final TypeAdapter<List<String>> java_util_List_java_lang_String_Adapter;\n"
+        + "    private final TypeAdapter<Map<String, Map<String, Map<String, Map<String, Map<String, String>>>>>> java_util_Map_java_lang_String__java_util_Map_java_lang_String__java_util_Map_java_lang_String__java_util_Map_java_lang_String__java_util_Map_java_lang_String__java_lang_String_____Adapter;\n"
```

Comment:
ones like this make me wonder if we should shorten package names to first letters or something, like dagger does

Reply:
I would say just drop the package name all together and use the simple name.  Most of the time that will be fine, and the `NameAllocator` will take care of collisions (so you'll have `StringAdapter` and `StringAdapter1`). And since you're storing them in a map by type, so the generator will be able to resolve the correct adapter.